### PR TITLE
Hazards

### PR DESCRIPTION
**Pull request by joshuajgit**
Ready to move hazards over to the rest of the project

### CODE REVIEW

File: Assets/CodeBase/DestroyThis.cs [hunk 1]
```diff
@@ -0,0 +1,16 @@
+﻿using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+/// <summary>
+/// Script to destroy object it is attached to after a specified delay in seconds.
+/// </summary>
+public class DestroyThis : MonoBehaviour
```
Comment (Megan-VanHumbeck) [inline Assets/CodeBase/DestroyThis.cs:8]: I don't really understand the necessity of this script. Could you please explain?

**Review by Megan-VanHumbeck — APPROVED** [review]

File: Assets/CodeBase/DestroyThis.cs (hunk 1 above)
Comment (DavidoBod) [reply Assets/CodeBase/DestroyThis.cs:8]: If it's to make the delay time universal (before deleting an object) I'd say just scrap it and input the delay manually in each class where those objects are being destroyed. But let me know if I'm missing the point of this class

File: Assets/CodeBase/Entities/Hazards/Landmine.cs [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+            detonating = true;
+            Invoke("Detonate", 0.5f);
+        }
+    }
+
+    public void Detonate() {
+        GameObject e = Instantiate(explosion, transform);
+        e.transform.localPosition = Vector3.zero;
+        e.transform.parent = null;
+        Destroy(this.gameObject);
```
Comment (DavidoBod) [inline Assets/CodeBase/Entities/Hazards/Landmine.cs:29]: We should kill the player too when they step on the landmine

Comment (JesseDes) [reply Assets/CodeBase/Entities/Hazards/Landmine.cs:29]: The explosion that it adds kills the player

**Review by JesseDes — APPROVED** [review]

**Pull request merged**